apps/files/src/components/FileEntry/FileEntryPreview.vue

```diff
@@ -206,24 +211,53 @@ export default Vue.extend({
 
 			return null
 		},
+
+		hasBlurhash() {
+			return this.source.attributes['metadata-blurhash'] !== undefined
+		},
+	},
+
+	mounted() {
+		if (this.hasBlurhash && this.$refs.canvas) {
+			this.drawBlurhash()
+		}
 	},
 
 	methods: {
 		// Called from FileEntry
 		reset() {
 			// Reset background state to cancel any ongoing requests
 			this.backgroundFailed = undefined
+			this.backgroundLoaded = false
 			if (this.$refs.previewImg) {
 				this.$refs.previewImg.src = ''
 			}
 		},
 
+		onBackgroundLoad(event) {
+			this.backgroundFailed = false
+			this.backgroundLoaded = true
+		},
+
 		onBackgroundError(event) {
 			// Do not fail if we just reset the background
 			if (event.target?.src === '') {
 				return
 			}
 			this.backgroundFailed = true
+			this.backgroundLoaded = false
+		},
+
+		drawBlurhash() {
+			const width = this.$refs.canvas.width
+			const height = this.$refs.canvas.height
+
+			const pixels = decode(this.source.attributes['metadata-blurhash'], width, height)
+
+			const ctx = this.$refs.canvas.getContext('2d')
+			const imageData = ctx.createImageData(width, height)
+			imageData.data.set(pixels)
+			ctx.putImageData(imageData, 0, 0)
```

Comment:
There are some type issues, so I would recommend to first cast the canvas ref properly and also check for null on `getContext` (this is a valid return value):

```suggestion
		drawBlurhash() {
			const canvas = this.$refs.canvas as HTMLCanvasElement
			const width = canvas.width
			const height = canvas.height

			const pixels = decode(this.source.attributes['metadata-blurhash'], width, height)

			const ctx = canvas.getContext('2d')
			if (ctx === null) {
				logger.error('Cannot create context for blurhash canvas')
				return
			}
			const imageData = ctx.createImageData(width, height)
			imageData.data.set(pixels)
			ctx.putImageData(imageData, 0, 0)
```